### ISSUE TITLE
Implement Proposals Widget

### ISSUE THREAD

**Issue by mmbotelho** [okTurtles/group-income#564]
_**Note**: I would recommend finishing issues #554, #560, #553 before taking this one. There might also be dependencies with the open PR #425 (Proposals / Voting logic)._

## Problem
![Captura de ecrã 2019-04-29, às 15 41 (1)](https://user-images.githubusercontent.com/22004566/56901233-f97f8280-6a97-11e9-8acd-26afbadafa2e.png)
- There's currently no way for the user to see archived proposals (related to issue #426);
- There's now the possibility to add a description to all proposals - this is not supported by the current UI;
- Now there's the possibility for a user to propose adding more than one member at a time (with one description) and the current UI does not support this (see #553);

## Solution
Contributions widget:
<img width="800" alt="Captura de ecrã 2019-04-29, às 16 22 39" src="https://user-images.githubusercontent.com/22004566/56903354-6dbc2500-6a9c-11e9-9ff5-6d14a02031fa.png">

Proposals page:
<img width="1640" alt="17  Dashboard" src="https://user-images.githubusercontent.com/22004566/56903489-b378ed80-6a9c-11e9-93ef-cd9d959d2be3.png">

- Implement the UI and copy changes made to the widget;
- Implement the "All Proposals" page;
- Add the possibility for a proposal to have a description;
- Make it possible to group "add a member" proposals;

Figma links:
- [Detailed widget mockups](https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN/Group-Income-2.0?node-id=1579%3A202);
- [Proposals page mockups](https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN/Group-Income-2.0?node-id=1988%3A1261);
- [Widget in the Dashboard](https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN/Group-Income-2.0?node-id=1257%3A344).

**Comment by mmbotelho** [comment]
FYI @sandrina-p I have updated the Figma links on the issue!

**Comment by sandrina-p** [comment]
@greg At the moment, when we create 1 proposal with 2 member invitations, 2 different proposals are created in the group state. However, they should be displayed together (https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN/Group-Income-2.0?node-id=1579%3A202).
So, we need to create somehow a “link” between these proposals so they can be grouped.
Do you have any suggestions about the best way we could approach that? I have some ideas but none of them are clean IMO...

To refresh your mind, at the moment this is the current object shape of `group.proposals`:

```js
{
  [proposalHash1]: {
    data: {
      expires_date_ms: Number,
      proposalData: {
        members: [String] // Array of usernames. Maybe the solution is here?
        reason: String
      },
      proposalType: String,
      votingRule: String,
    },
    meta: {},
    payload: Object // when is an invite.
    status: String,
    votes: {
      [username]: String
    }
  },
  [proposalHash2]: { ... }
}
```

**Comment by taoeffect** [comment]
@sandrina-p 

> So, we need to create somehow a “link” between these proposals so they can be grouped.
Do you have any suggestions about the best way we could approach that? I

~~They are already linked: the way @mmbotelho designed it in the link you shared is that they're linked by the person who submitted the proposal. So in the UI, group proposals together based on who submitted them.~~

**Comment by sandrina-p** [comment]
Hum.. it should be created by the same person and at the same time. So, if the same user creates two proposals in different days, those proposals should not be grouped

**Comment by taoeffect** [comment]
@sandrina-p - actually, ignore my comment above, we're hashing the details out right now over Slack

**Comment by sandrina-p** [comment]
Closed by #698
